### PR TITLE
Update docs/HowSwiftImportsCAPIs.md

### PR DESCRIPTION
**Pull request by benrimmington**
* Increase heading levels
* Update the introduction
* Use `Int` in `qsort` examples
* Use `CInt` in anonymous structs
* Use `CUnsignedInt` in enums
* De-dupe `HomeworkExcuse` enum

### CODE REVIEW

**Comment by benrimmington** [comment]
Cc: @gribozavr

File: docs/HowSwiftImportsCAPIs.md [hunk 1]
```diff
@@ -929,30 +929,30 @@ enum CardinalDirection {
...
-struct HomeworkExcuse : Equatable, RawRepresentable {
-  init(_ rawValue: UInt32)
-  init(rawValue: UInt32)
-  var rawValue: UInt32
-  typealias RawValue = UInt32
+struct HomeworkExcuse: Equatable, RawRepresentable {
+  init(_ rawValue: CUnsignedInt)
+  init(rawValue: CUnsignedInt)
+  var rawValue: CUnsignedInt { get }
+  typealias RawValue = CUnsignedInt
```
Comment (benrimmington) [inline docs/HowSwiftImportsCAPIs.md]: @gribozavr Can this repeated `HomeworkExcuse` example be removed?

File: docs/HowSwiftImportsCAPIs.md [hunk 2]
```diff
@@ -929,30 +929,30 @@ enum CardinalDirection {
...
-@_frozen
-enum CardinalDirection : UInt32 {
-  init?(rawValue: UInt32)
-  var rawValue: UInt32 { get }
-  typealias RawValue = UInt32
+@frozen
+enum CardinalDirection: CUnsignedInt, Hashable, RawRepresentable {
+  init?(rawValue: CUnsignedInt)
+  var rawValue: CUnsignedInt { get }
+  typealias RawValue = CUnsignedInt
```
Comment (benrimmington) [inline docs/HowSwiftImportsCAPIs.md]: @gribozavr Can these implicit conformances be removed?

```swift
enum InputDevice: CUnsignedInt {
  case Keyboard
  case Mouse
  case Touchscreen
}

@frozen
enum CardinalDirection: CUnsignedInt {
  case East
  case West
  case North
  case South
}
```

**Review by gribozavr — APPROVED** [review]
Thanks!

File: docs/HowSwiftImportsCAPIs.md [hunk 3]
```diff
@@ -929,30 +922,14 @@ enum CardinalDirection {
...
-}
-var EatenByPet: HomeworkExcuse { get }
-var ForgotAtHome: HomeworkExcuse { get }
-var ThoughtItWasDueNextWeek: HomeworkExcuse { get }
-
-enum InputDevice : UInt32 {
-  init?(rawValue: UInt32)
-  var rawValue: UInt32 { get }
-  typealias RawValue = UInt32
+enum InputDevice: CUnsignedInt {
```
Comment (gribozavr) [inline docs/HowSwiftImportsCAPIs.md]: I would prefer to include all conformances to show the complete API that is available.

**Comment by benrimmington** [comment]
@gribozavr I've squashed the commits, but I haven't got permission to test and merge.

**Comment by gribozavr** [comment]
@swift-ci Please test and merge.

**Comment by gribozavr** [comment]
@swift-ci Please smoke test and merge

**Comment by benrimmington** [comment]
Test and smoke test failed:

* <https://ci.swift.org/job/swift-PR-merge/2251/>

* <https://ci.swift.org/job/swift-PR-smoke-merge/4203/>

**Comment by theblixguy** [comment]
@swift-ci please smoke test and merge

**Pull request merged**